tests/int/run.py

```diff
@@ -1,317 +1,282 @@
+
+  def test_complex_range_with_three_segments_and_spaces(self):
+    for pair in [(0, True), (13, False), (-20, False), (5, True), (16, True)]:
+      with self.subTest(pair=pair):
+        allowed = True
+        try:
+          self.int_obj.int_container.complex_range_two = pair[0]
+        except ValueError:
+          allowed = False
+        self.assertEqual(allowed, pair[1],
+          "Complex range with spaces and three elements was not set correctly (%d -> %s != %s)" %
+          (pair[0], allowed, pair[1]))
+
+  def test_complex_range_with_negative(self):
+    for pair in [(-2, True), (42, False)]:
+      with self.subTest(pair=pair):
+        allowed = True
+        try:
+          self.int_obj.int_container.complex_range_with_negative = pair[0]
+        except ValueError:
+          allowed = False
+        self.assertEqual(allowed, pair[1], "Complex range with negatives was not set correctly (%d -> %s != %s)" %
+          (pair[0], allowed, pair[1]))
+
+  def test_int8_range_with_negatives_and_spaces(self):
+    for pair in [(-11, False), (-5, True), (0, False), (5, False), (10, True),
+                 (25, True), (31, False)]:
+      with self.subTest(pair=pair):
+        allowed = True
+        try:
+          self.int_obj.int_container.intLeafWithRange = pair[0]
+        except ValueError:
+          allowed = False
+        self.assertEqual(allowed, pair[1],
+          "Complex range with negatives and additional spaces was not set correctly (%d -> %s != %s)" %
+          (pair[0], allowed, pair[1]))
+
+  def test_complex_range_with_equals_case(self):
+    for pair in [(-43, False), (-40, True), (98, False), (122, True), (254, False),
+                 (255, True), (256, False)]:
+      with self.subTest(pair=pair):
+        allowed = True
+        try:
+          self.int_obj.int_container.complex_range_with_equals_case = pair[0]
+        except ValueError:
+          allowed = False
+        self.assertEqual(allowed, pair[1],
+          "Complex range with equals case was not set correctly (%d -> %s != %s)" %
+          (pair[0], allowed, pair[1]))
+
+  def test_setters_exist(self):
+    for leaf in ["eight", "sixteen", "thirtytwo", "sixtyfour"]:
+      with self.subTest(leaf=leaf):
+        setter = getattr(self.int_obj.int_container, "_set_%s" % leaf, None)
+        self.assertIsNotNone(setter, "Could not find attribute setter for %s" % leaf)
+
+  def test_set_int_sizes_at_lower_bounds(self):
+    bounds = {
+      'eight': -2**7,
+      'sixteen': -2**15,
+      'thirtytwo': -2**31,
+      'sixtyfour': -2**63,
+    }
+    for leaf, value in bounds.iteritems():
+      with self.subTest(leaf=leaf):
+        setter = getattr(self.int_obj.int_container, "_set_%s" % leaf)
+        allowed = True
+        try:
+          setter(value)
+        except ValueError:
+          allowed = False
+        self.assertTrue(allowed, "Could not set int size %s to %d" % (leaf, value))
+
+  def test_set_int_sizes_below_lower_bounds(self):
```

Comment:
Looks like I wrote this the same way as this in the original test, but now you redeclare `bounds`, I recommend moving the `value - 1` from line 209 to just specifying it within the `bounds` dict. This is done in the next test down and it seems cleaner.